dev/com.ibm.ws.classloading/src/com/ibm/wsspi/classloading/ApiType.java

```diff
@@ -61,15 +67,115 @@ public static ApiType fromString(String value) {
     /** Convert one or more comma-and-space-delimited api type strings into a single set of types */
     public static EnumSet<ApiType> createApiTypeSet(String... apiTypes) {
         EnumSet<ApiType> set = EnumSet.noneOf(ApiType.class);
+        EnumSet<ApiType> removeSet = EnumSet.noneOf(ApiType.class);
+        EnumSet<ApiType> addSet = EnumSet.noneOf(ApiType.class);
+        ArrayList<String> notFoundSet = new ArrayList<String>();
+        StringBuffer initialtypes = new StringBuffer();
         if (apiTypes != null)
-            for (String types : apiTypes)
+            for (String types : apiTypes) {
+                if (initialtypes.length() == 0)
+                    initialtypes.append("\"" + types + "\"");
+                else
+                    initialtypes.append(" \"" + types + "\"");
                 if (types != null)
                     for (String stype : types.split("[ ,]+")) {
-                        ApiType type = ApiType.fromString(stype);
-                        if (type != null) {
-                            set.add(type);
+                        if (stype.indexOf("+") == 0) {
+                            stype = stype.replaceFirst("\\+", "");
+                            ApiType type = ApiType.fromString(stype);
+                            if (type != null) {
+                                if (!addSet.add(type)) {
+                                    if (tc.isErrorEnabled())
```

Comment:
it isn't necessary to check if error is enabled before issuing a `Tr.error()`

Reply:
I was following the code conventions of  similar code for classloading.   Not sure why the have tc.ErrorEnabled, but it likely does not hurt, so, I think I will leave it.